records_mover/airflow/hooks/records_hook.py

```diff
@@ -28,7 +28,9 @@ def _get_boto3_session(self) -> boto3.session.Session:
 
     @property
     def _url_resolver(self) -> UrlResolver:
-        return UrlResolver(boto3_session=self._get_boto3_session())
+        return UrlResolver(boto3_session_getter=self._get_boto3_session,
```

Comment:
Up until this PR, we had been eagerly fetching the Boto3 Session object.  That's been fine, as it's a pretty fast thing to fetch, usually.

Unfortunately, that is not true of GCP creds, at least in Application Default Credentials form.  There's a significant delay as they are returned, and an obnoxious "we've deprecated this functionality but have no realistic replacement" log message, to boot!

As a result, I put in machinery to both load this lazily only when we're about and cache it.  I applied the same machinery to the boto3 credentials for consistency's sake.

Much of this responsibility lives in the UrlResolver class, which is responsible for the caching and lazy loading.  The Session object (and this RecordsHook) are responsible for passing in a function that returns the creds (or None if none are configured) when called.